### PR TITLE
Upgrade to nextv12 and routing update to next/router

### PR DESCRIPTION
**Pull request by MartinTrojans**
Signed-off-by: Ding Ma <13991523+MartinTrojans@users.noreply.github.com>

### CODE REVIEW

File: ui/next.config.js [hunk 1]
```diff
@@ -13,11 +13,3 @@
  * See the License for the specific language governing permissions and
```
Comment (abvaidya) [inline ui/next.config.js:13]: lets remove this file if not being used

File: ui/package.json [hunk 2]
```diff
@@ -70,25 +70,23 @@
         "moment": "2.29.1",
         "moment-timezone": "0.5.33",
         "multer": "1.4.3",
-        "next": "10.2.3",
-        "next-compose-plugins": "2.2.1",
-        "next-routes": "1.4.2",
+        "next": "^12.0.4",
```
Comment (abvaidya) [inline ui/package.json:73]: we decided to use exact versions ( no carrot symbol ) to not break builds on-prem

File: ui/src/__tests__/components/utils/JsonUtils.test.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2020 Verizon Media
```
Comment (abvaidya) [inline ui/src/__tests__/components/utils/JsonUtils.test.js:2]: For new files we are using The Athenz Authors in copyright text

File: ui/src/pages/_document.js [hunk 4]
```diff
@@ -16,25 +16,25 @@
...
-                        data-emotion-css={styles.ids.join(' ')}
-                        nonce={ctx.req.headers.rid}
-                        dangerouslySetInnerHTML={{ __html: styles.css }}
-                    />
-                </>
-            ),
-        };
-    }
+export async function getServerSideProps(ctx) {
+    const initialProps = await Document.getInitialProps(ctx);
```
Comment (abvaidya) [inline ui/src/pages/_document.js:20]: Should this be Document.getServerSideProps?

File: ui/src/components/utils/JsonUtils.js [hunk 5]
```diff
@@ -14,5 +14,12 @@
  * limitations under the License.
  */
 
-import RolePage from './role';
-export default RolePage;
+export default class JsonUtils {
```
Comment (abvaidya) [inline ui/src/components/utils/JsonUtils.js:17]: For new files lets use the new "The Athenz Authors" copyright

**Review by abvaidya — CHANGES_REQUESTED** [review]

File: ui/src/pages/_document.js (hunk 4 above)
Comment (MartinTrojans) [reply ui/src/pages/_document.js:20]: Document object provided by next doesn't have getServerSideProps.

**Review by abvaidya — APPROVED** [review]

**Comment by abvaidya** [comment]
closes #1607

**Pull request merged**